Original patch from Gentoo.

-= BEGIN original header =-
-= END original header =-

diff -durN uClibc-0.9.30.orig/libc/sysdeps/linux/arm/bits/uClibc_arch_features.h uClibc-0.9.30/libc/sysdeps/linux/arm/bits/uClibc_arch_features.h
--- uClibc-0.9.30.orig/libc/sysdeps/linux/arm/bits/uClibc_arch_features.h	2008-09-25 10:50:56.000000000 +0200
+++ uClibc-0.9.30/libc/sysdeps/linux/arm/bits/uClibc_arch_features.h	2009-02-07 09:57:59.000000000 +0100
@@ -39,4 +39,8 @@
 /* define if target supports IEEE signed zero floats */
 #define __UCLIBC_HAVE_SIGNED_ZERO__
 
+#ifdef __ARM_EABI__
+# define __UCLIBC_TRUNCATE64_HAS_4_ARGS__
+#endif
+
 #endif /* _BITS_UCLIBC_ARCH_FEATURES_H */
